lib/ui/settings/security_section_widget.dart

```diff
@@ -129,17 +125,25 @@ class _SecuritySectionWidgetState extends State<SecuritySectionWidget> {
             Switch.adaptive(
               value: _config.shouldShowLockScreen(),
               onChanged: (value) async {
-                AppLock.of(context).disable();
-                final result = await requestAuthentication(
-                  "Please authenticate to change lockscreen setting",
-                );
-                if (result) {
-                  AppLock.of(context).setEnabled(value);
-                  _config.setShouldShowLockScreen(value);
-                  setState(() {});
+                if (await LocalAuthentication().isDeviceSupported()) {
```

Comment:
Could you explain why? The service would be something that most probably won't be used anywhere else. So I'm assuming you want to move it because there will be less lines of code on this block?

Reply:
Yes, that, plus it'll be nice if all calls to the `LocalAuthentication` package are routed through a single proxy (`LocalAuthenticationService`). So if tomorrow we've to replace this dependency, changes would be contained to a single file.